django_afip/models.py

```diff
@@ -1502,10 +1509,42 @@ class ReceiptEntry(models.Model):
         on_delete=models.PROTECT,
     )
 
+    def _validate_discount(self) -> exceptions.DjangoAfipException | None:
+        """
+        This method checks the discount value with two conditions assuming that
+        total price before discount is quantity * unit_price and that
+        total price is quantity * unit_price - discount.
+
+        - `discount` must be a positive value due to the fact that
+            if discount is a negative value, that will cause an increment
+            in the total price.
+        - `discount` must be less than or equal to total price because if discount is
+            greater than total price before discount that will cause total_price
+            to be negative.
+
+        if this validation check fails, a DjangoAfipException will be raised.
+        """
+
+        if self.discount > self.quantity * self.unit_price:
+            msg = "discount must be less than or equal to total price before discount"
+            raise exceptions.DjangoAfipException(_(msg))
```

Comment:
I think it's best to use a [constraint check](https://docs.djangoproject.com/en/4.1/ref/models/constraints/#check) to validate this, so the DB rejects invalid values. E.g.:

```py
CheckConstraint(
    check=Q(discount__lte=F("quantity") * F("unit_price")),
    name='discount_less_than_total'
) 
```

I haven't tried this though; if it has issues we can postpone it and leave this check for now.